arbnode/blockmetadata.go

```diff
@@ -131,31 +134,35 @@ func (b *BlockMetadataFetcher) fetch(ctx context.Context, fromBlock, toBlock uin
 	return result, nil
 }
 
-func (b *BlockMetadataFetcher) persistBlockMetadata(ctx context.Context, query []uint64, result []gethexec.NumberAndBlockMetadata) error {
+func (b *BlockMetadataFetcher) persistBlockMetadata(ctx context.Context, query []uint64, result []gethexec.NumberAndBlockMetadata) (bool, error) {
 	batch := b.db.NewBatch()
 	queryMap := util.ArrayToSet(query)
 	for _, elem := range result {
+		if (b.syncUntilBlock != 0) && (elem.BlockNumber > b.syncUntilBlock) {
+			b.StopWaiterSafe.StopOnly()
+			return true, nil
```

Comment:
Since this calls `StopOnly`, why does it also need to return `done`? Stopping the StopWaiter should prevent Update from being called further.

Reply:
Josh explained to me that it's to terminate the loop in Update early- could that loop check StopWaiter.Stopped() instead?

Reply:
done